src/test/headers_sync_chainwork_tests.cpp

```diff
@@ -15,9 +15,36 @@
 
 #include <boost/test/unit_test.hpp>
 
+using State = HeadersSyncState::State;
+
 constexpr size_t TARGET_BLOCKS{15'000};
 constexpr arith_uint256 CHAIN_WORK{TARGET_BLOCKS * 2};
 
+// Standard set of checks common to all scenarios. Macro keeps failure lines at the call-site.
+#define CHECK_RESULT(result_expression, hss, exp_state, exp_success, exp_request_more,                     \
+                     exp_headers_size, exp_pow_validated_prev, exp_locator_hash)                           \
+    do {                                                                                                   \
+        const auto result = result_expression;                                                             \
+        BOOST_REQUIRE_EQUAL(hss.GetState(), exp_state);                                                    \
+        BOOST_CHECK_EQUAL(result.success, exp_success);                                                    \
+        BOOST_CHECK_EQUAL(result.request_more, exp_request_more);                                          \
+        BOOST_CHECK_EQUAL(result.pow_validated_headers.size(), exp_headers_size);                          \
+        std::optional<uint256> locator_hash_opt{exp_locator_hash};                                         \
+        if (locator_hash_opt == std::nullopt) {                                                            \
+            BOOST_CHECK_EQUAL(exp_state, State::FINAL);                                                    \
+        } else {                                                                                           \
+            BOOST_REQUIRE_GT(hss.NextHeadersRequestLocator().vHave.size(), 0);                             \
```

Comment:
would't an `.at(0)` suffice?

Reply:
I think it applies to `BOOST_REQUIRE_GT(result.pow_validated_headers.size(), 0);` as well